File: backend/workspace/temp/code-review/65ed3813-c87f-4a61-8fc4-ddf745e8125b/repo/src/main/java/com/mickey/onlineordering/onlineorderingserver/entity/Address.java
package com.mickey.onlineordering.onlineorderingserver.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 收货地址实体类
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Address {
    private Long id;
    private Long userId;
    private String receiverName;
    private String receiverPhone;
    private String address;
    private Boolean isDefault;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: backend/workspace/temp/code-review/65ed3813-c87f-4a61-8fc4-ddf745e8125b/repo/online-ordering-client/src/api/address.ts
import request from '@/utils/request'

export interface Address {
  id?: number
  userId?: number
  receiverName: string
  receiverPhone: string
  address: string
  isDefault: boolean
  createdAt?: string
  updatedAt?: string
}

// 获取地址列表
export function getAddressList() {
  return request({
    url: '/addresses',
    method: 'get'
  })
}

// 获取地址详情
export function getAddressDetail(id: number) {
  return request({
    url: `/addresses/${id}`,
    method: 'get'
  })
}

// 添加地址
export function addAddress(data: Address) {
  return request({
    url: '/addresses',
    method: 'post',
    data
  })
}

// 更新地址
export function updateAddress(id: number, data: Address) {
  return request({
    url: `/addresses/${id}`,
    method: 'put',
    data
  })
}

// 删除地址
export function deleteAddress(id: number) {
  return request({
    url: `/addresses/${id}`,
    method: 'delete'
  })
}

// 设置默认地址
export function setDefaultAddress(id: number) {
  return request({
    url: `/addresses/${id}/default`,
    method: 'put'
  })
}

